src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs

```diff
@@ -64,7 +65,23 @@ await using (childScope.ConfigureAwait(false))
                 throw;
             }
 
-            await receivePipelineNotification.Raise(new ReceivePipelineCompleted(message, pipelineStartedAt, DateTimeOffset.UtcNow), cancellationToken).ConfigureAwait(false);
+            var completedAt = DateTimeOffset.UtcNow;
+            var incomingPipelineMetricTags = transportReceiveContext.Extensions.Get<IncomingPipelineMetricTags>();
+            if (incomingPipelineMetricTags.IsMetricTagsCollectionEnabled)
+            {
+                TagList tags;
+                incomingPipelineMetricTags.ApplyTags(ref tags, [
+                    MeterTags.QueueName,
+                    MeterTags.EndpointDiscriminator,
+                    MeterTags.MessageType]);
```

Comment:
@lailabougria @SzymonPobiega, is it useful to add here the `MeterTags.MessageHandlerTypes`? It'll be published to both critical and processing time, reporting all the handlers involved./

```suggestion
                incomingPipelineMetricTags.ApplyTags(ref tags, [
                    MeterTags.QueueName,
                    MeterTags.EndpointDiscriminator,
                    MeterTags.MessageType,
                    MeterTags.MessageHandlerTypes]);
```